frontend/src/helpers/api.js

```diff
@@ -16,11 +18,16 @@ class Api {
             ...headers
         };
 
-        return fetch(requestURL, {
+        const options = {
             method: action,
-            headers: requestHeaders,
-            body: JSON.stringify(payload)
-        }).then(res => res.json()).catch(e => e);
+            headers: requestHeaders
+        }
+
+        if (!_.isEmpty(payload)) {
```

Comment:
I'm not sold that we need to bring in a dependency just to check if a data structure is empty. If we do need a utility library, I do recommend `lodash` since it's tree-shakeable and we can just bundle the used functions.